tests/src/python/test_qgsgrouplayer.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for QgsGroupLayer.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Nyall Dawson'
+__date__ = '22/11/2021'
+__copyright__ = 'Copyright 2021, The QGIS Project'
+
+import qgis  # NOQA
+
+import tempfile
+import os
+import gc
+from qgis.PyQt.QtCore import (
+    QCoreApplication,
+    QEvent,
+    QSize,
+    QDir
+)
+from qgis.PyQt.QtGui import (
+    QPainter,
+    QColor
+)
+from tempfile import TemporaryDirectory
+
+from qgis.core import (
+    QgsFeature,
+    QgsVectorLayer,
+    QgsProject,
+    QgsCoordinateTransformContext,
+    QgsGroupLayer,
+    QgsGeometry,
+    QgsPointXY,
+    QgsMapSettings,
+    QgsMultiRenderChecker,
+    QgsDropShadowEffect,
+    QgsEffectStack,
+    QgsDrawSourceEffect
+)
+from qgis.testing import start_app, unittest
+from utilities import unitTestDataPath
+
+start_app()
+
+TEST_DATA_DIR = unitTestDataPath()
+
+
+class TestQgsGroupLayer(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.report = "<h1>Python QgsGroupLayer Tests</h1>\n"
+
+    @classmethod
+    def tearDownClass(cls):
+        report_file_path = "%s/qgistest.html" % QDir.tempPath()
+        with open(report_file_path, 'a') as report_file:
+            report_file.write(cls.report)
+
+    def test_children(self):
+        options = QgsGroupLayer.LayerOptions(QgsCoordinateTransformContext())
+        group_layer = QgsGroupLayer('test', options)
+        self.assertTrue(group_layer.isValid())
+
+        self.assertFalse(group_layer.childLayers())
+
+        # add some child layers
+        layer1 = QgsVectorLayer('Point?crs=epsg:3111', 'Point', 'memory')
+        layer2 = QgsVectorLayer('Point?crs=epsg:4326', 'Point', 'memory')
+
+        group_layer.setChildLayers([layer1, layer2])
```

Comment:
API wise, would it be useful to add an appendChildLayers(), or we're fine with people going layers = group_layer.childLayers(); layers += [more,layers] ?

Reply:
I'll do that in a follow up